connect.go

```diff
@@ -89,7 +89,7 @@ type connectDialer struct {
 // newConnectDialer creates a dialer to issue CONNECT requests and tunnel traffic via HTTP/S proxy.
 // proxyUrlStr must provide Scheme and Host, may provide credentials and port.
 // Example: https://username:password@golang.org:443
-func newConnectDialer(proxyUrlStr string, timeout time.Duration) (proxy.ContextDialer, error) {
+func newConnectDialer(proxyUrlStr string, timeout time.Duration, localAddr *net.TCPAddr) (proxy.ContextDialer, error) {
```

Comment:
i think we want this also on `func newDirectDialer(timeout time.Duration) proxy.ContextDialer` which is used when no proxyUrl is used.

Reply:
@NikosEfthias got it thank you. Then i assume we want this option also on the directDialer (and maybe even von the Socks dialer?). Currently your implementation sets this option only on the proxy dialer